cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -85,19 +85,31 @@ type sfdiskPartition struct {
 	Name  string `json:"name"`
 }
 
-func (p *sfdiskPartition) isCreated() bool {
-	// TODO:UC20: also provide a mechanism for MBR (RPi)
-	if !creationSupported(p.Type) {
-		return false
-	}
-	for _, a := range strings.Fields(p.Attrs) {
-		if !strings.HasPrefix(a, "GUID:") {
-			continue
+func isCreatedDuringInstall(p *sfdiskPartition, fs *lsblkBlockDevice, sfdiskLabel string) bool {
+	switch sfdiskLabel {
+	case "gpt":
+		// the created partitions use specific GPT GUID types and set a
+		// specific bit in partition attributes
+		if !creationSupported(p.Type) {
+			return false
 		}
-		attrs := strings.Split(a[5:], ",")
-		if strutil.ListContains(attrs, createdPartitionAttr) {
-			return true
+		for _, a := range strings.Fields(p.Attrs) {
+			if !strings.HasPrefix(a, "GUID:") {
+				continue
+			}
+			attrs := strings.Split(a[5:], ",")
+			if strutil.ListContains(attrs, createdPartitionAttr) {
+				return true
+			}
 		}
+	case "dos":
+		// we have no similar type/bit attribute setting for MBR, on top
+		// of that MBR does not support partition names, fall back to
+		// reasonable assumption that only partitions carrying
+		// ubuntu-boot and ubuntu-data labels are created during
+		// install, everything else was part of factory image
+		createdDuringInstall := []string{ubuntuBootLabel, ubuntuDataLabel}
+		return strutil.ListContains(createdDuringInstall, fs.Label)
```

Comment:
should we add a TODO about considering using the gadget offsets here at some point?